server/controllers/finance/stepDownCostAllocationQuantity.js

```diff
@@ -166,3 +168,87 @@ function remove(req, res, next) {
     })
     .catch(next);
 }
+
+// Update computable allocation basis quantities
+//
+// UPDATE /cost_center_allocation_basis_quantities_update
+//
+
+async function updateQuantities(req, res, next) {
+
+  // Get the full list of cost center IDs
+  const costCentersQuery = 'SELECT id, label AS name from cost_center';
+  const costCenters = await db.exec(costCentersQuery);
+
+  // Get the allocation bases that are computable
+  const cabQuery = 'SELECT id, name FROM cost_center_allocation_basis WHERE is_computed = 1';
+  const computables = await db.exec(cabQuery);
+
+  // Queries to update the quantity records
+  const findQRec = 'SELECT id FROM `cost_center_allocation_basis_value` WHERE `cost_center_id` = ? AND `basis_id` = ?';
+  const updateQRec = 'UPDATE `cost_center_allocation_basis_value` SET `quantity` = ? WHERE `id` = ?';
+  const insertQRec = 'INSERT INTO `cost_center_allocation_basis_value` (`cost_center_id`, `basis_id`, `quantity`) VALUES (?, ?, ?)';
+
+  // Compute and set each computable allocation basis quantity
+  const unUpdatedCenters = [];
+
+  /* eslint-disable no-await-in-loop */
+  for (let i = 0; i < computables.length; i++) {
+    const basis = computables[i];
+    const data = await allocationQuantities(basis.id);
+
+    // Update quantities for each cost center
+    for (let k = 0; k < costCenters.length; k++) {
+      const cc = costCenters[k];
+
+      // Get the data for this cost center (if any)
+      const ccData = data.find(item => item.cost_center_id === cc.id);
+      if (ccData) {
+        const newQuantity = ccData[basis.name];
+        // See if there is an existing quantity
+        const qRecordId = await db.exec(findQRec, [cc.id, basis.id]);
+        if (qRecordId.length > 0) {
```

Comment:
Can we just delete these values and rewrite?  I.e. if we are computing the number of employees, just clear out the old values and then do an INSERT for the rest?